octodns/provider/rackspace.py

```diff
@@ -0,0 +1,385 @@
+#
+#
+#
+from __future__ import absolute_import, division, print_function, \
+    unicode_literals
+
+from requests import HTTPError, Session, post
+from collections import defaultdict
+import logging
+import string
+import time
+
+from ..record import Record
+from .base import BaseProvider
+
+
+def add_trailing_dot(s):
+    assert s
+    assert s[-1] != '.'
+    return s + '.'
+
+
+def remove_trailing_dot(s):
+    assert s
+    assert s[-1] == '.'
+    return s[:-1]
+
+
+def escape_semicolon(s):
+    assert s
+    return string.replace(s, ';', '\;')
+
+
+def unescape_semicolon(s):
+    assert s
+    return string.replace(s, '\;', ';')
+
+
+class RackspaceProvider(BaseProvider):
+    SUPPORTS_GEO = False
+    SUPPORTS = set(('A', 'AAAA', 'ALIAS', 'CNAME', 'MX', 'NS', 'PTR', 'SPF',
+                    'TXT'))
+    TIMEOUT = 5
+
+    def __init__(self, id, username, api_key, ratelimit_delay, *args,
```

Comment:
Should `ratelimit_delay` have a default? The example config below doesn't specify a value for it, but it appears to be a required param here.